apps/android/app/src/main/java/com/axon/app/ui/ask/AskGeneration.kt

```
@@ -0,0 +1,145 @@
+package com.axon.app.ui.ask
+
+import android.os.SystemClock
+import androidx.lifecycle.viewModelScope
+import com.axon.app.data.local.AskHistoryEntry
+import com.axon.app.data.remote.AskStreamEvent
+import com.axon.app.data.repository.AskResultUi
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.launch
+
+internal fun AskViewModel.askFromQuery(query: String, attachment: String? = null) {
```

detekt (1.23.8)

[warning] The function askFromQuery appears to be too complex based on Cyclomatic Complexity (complexity: 25). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CyclomaticComplexMethod)